components/stripe/actions/cancel-payment-intent/cancel-payment-intent.js

```diff
@@ -0,0 +1,34 @@
+const pick = require("lodash.pick");
+const stripe = require("../../stripe.app.js");
+
+module.exports = {
+  key: "stripe-cancel-payment-intent",
+  name: "Cancel a Payment Intent",
+  version: "0.0.1",
+  description: "Cancel a [payment intent](https://stripe.com/docs/payments/payment-intents). " +
+    "Once canceled, no additional charges will be made by the payment intent and any operations " +
+    "on the payment intent will fail with an error. For payment intents with status=" +
+    "`requires_capture`, the remaining amount_capturable will automatically be refunded.",
+  props: {
+    stripe,
+    id: {
+      propDefinition: [
+        stripe,
+        "payment_intent",
+      ],
+      required: true,
+    },
+    cancellation_reason: {
```

Comment:
Is this optional? If so you'll want to add `optional: true` either here or in the prop definition in the app file.

Reply:
The UI doesn't yet communicate prop optionality well, and that's on the backlog.